### PR TITLE
Adding eDifFIQA(T) a light-weight model for face image quality assessment.

### PR DESCRIPTION
**Pull request by LSIbabnikz**
The pull request adds a new face image quality assessment method to the OpenCV ZOO library. The implemented model is based on a mobilefacenet backbone for efficiency, while maintaining the superb results (of the larger models) presented in the original [paper](https://lmi.fe.uni-lj.si/wp-content/uploads/2024/03/TBIOM___DifFIQAv2.pdf).

### CODE REVIEW

File: models/face_image_quality_assessment_ediffiqa/demo.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+# It is subject to the license terms in the LICENSE file found in the same directory.
+
+
+import sys
+import argparse
+
+import numpy as np
+import cv2 as cv
+
+from skimage import transform as trans
```
Comment (fengyuentau) [inline models/face_image_quality_assessment_ediffiqa/demo.py:11]: Could it be replaced with opencv functions?

Basically we use `opencv` for any image operations and `numpy` for array operations.

File: models/face_image_quality_assessment_ediffiqa/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# eDifFIQA(T)
+
+eDifFIQA(T) is a light-weight version of the models presented in the paper [eDifFIQA: Towards Efficient Face Image Quality Assessment based on Denoising Diffusion Probabilistic Models](https://ieeexplore.ieee.org/document/10468647), it achieves state-of-the-art results in the field of face image quality assessment.
+
+Notes:
+
+- The original implementation can be found [here](https://github.com/LSIbabnikz/eDifFIQA).
+- The included model combines a pretrained MobileFaceNet backbone, with a quality regression head trained using the proceedure presented in the original paper.
+- The model predicts quality scores of aligned face samples, where a higher predicted score corresponds to a higher quality of the input sample.
```
Comment (fengyuentau) [inline models/face_image_quality_assessment_ediffiqa/README.md:9]: Could you add a description on the quality of face, e.g. how many the score it needs to reach to be a good quality face image.

File: models/face_image_quality_assessment_ediffiqa/ediffiqa.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# This file is part of OpenCV Zoo project.
+# It is subject to the license terms in the LICENSE file found in the same directory.
+#
+# Copyright (C) 2021, Shenzhen Institute of Artificial Intelligence and Robotics for Society, all rights reserved.
+# Third party copyrights are property of their respective owners.
```
Comment (fengyuentau) [inline models/face_image_quality_assessment_ediffiqa/ediffiqa.py:5]: Please drop these two lines as it is not valid.

**Review by fengyuentau — COMMENTED** [review]
Thank you for contribution 👍 Please take a look at the comments below.

File: models/face_image_quality_assessment_ediffiqa/ediffiqa.py (hunk 3 above)
Comment (LSIbabnikz) [reply models/face_image_quality_assessment_ediffiqa/ediffiqa.py]: I have removed the lines as requested.

File: models/face_image_quality_assessment_ediffiqa/README.md (hunk 2 above)
Comment (LSIbabnikz) [reply models/face_image_quality_assessment_ediffiqa/README.md:9]: It is difficult to define an exact score that correlates well with a high quality image. Instead, I have added an analysis using two datasets, XQLFW and LFW, comparing their quality score distributions. The distributions show the differences between the quality scores of mostly high quality images from LFW compared to mixed quality images from the XQLFW dataset.

File: models/face_image_quality_assessment_ediffiqa/demo.py (hunk 1 above)
Comment (LSIbabnikz) [reply models/face_image_quality_assessment_ediffiqa/demo.py]: The skimage function call was replaced using the appropriate OpenCV function call.  

`tfm, _ = cv.estimateAffinePartial2D(src_pts, ref_pts, method=cv.LMEDS)`

**Review by fengyuentau — APPROVED** [review]
LGTM 👍

**Pull request merged**